client/utils/utils.c

```diff
@@ -497,18 +497,19 @@ int validate_ipv4_port(const char *str)
 
 int validate_ascii(const char *str)
 {
-	if(!str || strlen(str) == 0) {
-		return 1;
+	if(str == NULL) {
+		return 1; // why?
```

Comment:
\0 is valid string.

Reply:
strlen(NULL) will say that there is some issue :D